cmd/mattermost/commands/team.go

```diff
@@ -300,3 +310,24 @@ func removeDuplicatesAndSortTeams(teams []*model.Team) []*model.Team {
 	})
 	return result
 }
+
+func archiveTeamCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	foundTeams := getTeamsFromTeamArgs(a, args)
+	for i, team := range foundTeams {
+		if team == nil {
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Team().Delete(team.Id, model.GetMillis()); result.Err != nil {
```

Comment:
Sure, This would avoid code change in `team_store`. Will address this and hopefully this should also resolve build failure issue as I'll revert the changes in `store` and `team_store`